### ISSUE TITLE
Feature request: Backblaze B2 support

### ISSUE THREAD

**Issue by kobuki** [s3ql/s3ql#45]
It would be nice to see support for B2. It's one of the newer ones but practically unbeatable in price, and we have success with it using other OSS backup tools like hashbackup or Restic. Docs are at https://www.backblaze.com/b2/docs/

**Comment by szepeviktor** [comment]
@kobuki See #8

**Comment by kobuki** [comment]
Thanks, I've stumbled upon that PR after I've opened this ticket. Sadly there's not much interest in B2 it seems.

**Comment by szepeviktor** [comment]
I've used that PR's code to back up my development server for ~6 months.

**Comment by kobuki** [comment]
That's certainly good news. What is there missing? Is it cosmetics or does it have showstopper issues? I might try the patch regardless.

**Comment by szepeviktor** [comment]
See the maintainer's requests in the PR!

**Comment by kobuki** [comment]
Yes, that's a load of small notes and requests, made like 1.5 years ago with no relevant response at all. I see no future in that but let's hope for the best.

**Comment by szepeviktor** [comment]
You may correct all of those, it may take 2 hours.

**Comment by kobuki** [comment]
Typing in the changes might take not a lot more but testing it as main building block of the FS would take considerably more. Plus, I don't know anything about the code structure so getting accustomed to its ways would require at least a few days of learning (and that's still sounds pretty steep). For the one responsible for the original PR it might not take more than a day, probably.

**Comment by kobuki** [comment]
I agree. It's sad that after @sylvainlehmann made a first version and obviously took a lot of effort in creating it, it never got finished and integrated. Maybe he got thwarted by the ~100 comments you made on his code?

**Comment by powerpaul17** [comment]
I am currently working on Backblaze B2 support for S3QL. Based on the existing PR and the comments I am preparing a new PR for a B2 backend. Should be ready in a few days.

**Comment by imperative** [comment]
Any news? A couple of months have passed. Are you still working on it?

**Comment by szepeviktor** [comment]
See https://github.com/s3ql/s3ql/pull/116

**Comment by kobuki** [comment]
PR #116 is not accepted. Are there any outstanding issues? Is it considered stable? This repo still looks abandoned, sadly.

**Comment by szepeviktor** [comment]
??? https://github.com/s3ql/s3ql/releases

**Comment by powerpaul17** [comment]
I'm using the backblaze backend (PR #116) on a daily basis, but I'm still waiting for a review from the maintainer..
